word-search.py

```diff
@@ -351,7 +353,52 @@ def new_game():  #merged and renamed word_search() into new_game()
         gridWordList = generate_word_list(valid_words, grid)
 
         foundWords = []
-        currentScore = 0  
+        currentScore = 0
+
+        # Timer integaration
+        game_duration  = timer(timeroption)
+        start_time = time.time() if game_duration else None
+        print("\nGame Start! Find words in the grid.")
+    
+        if game_duration: # Timer active options 1, 2, 3
+            print (f"Game Timer: {game_duration} seconds")
+        else:
+            print("\nGame is untimed!")
+
+        # Game loop
+        while True:
+            # Timer logic
+            if game_duration:
+                elapsed_time = time.time() -  start_time
+                remaining_time = game_duration - int(elapsed_time)
+                if remaining_time <= 0: # Time's up
+                    print("Time's up. Game Over!")
+                    break
+
+            print(f"\rTime remaining: {remaining_time} seconds", end="", flush=True)
+            time.sleep(1)
+
+            # Game Interaction     
+            wordInput = input('\nEnter word (or type "0" to quit): ')
+        
+            if wordInput.lower() == "0":
+                print(f"\nYour final score is: {currentScore}. \nThank you for playing!")
+                return
+
+            if wordInput in foundWords:
+                print("You've already found this word. Try another one!")
+            elif wordInput in gridWordList:
+                foundWords.append(wordInput)
+                score = scoreWord(wordInput)  
+                currentScore += score  
+                print(f"Valid word! Your current score is {currentScore}.")
+            else:
+                print("Invalid word. Try again.")
```

Comment:
Hi George, this part is repeating. Please combine with lines 403 - 432 below. The code below is the updated version for handling input options during the game. 🙂

Reply:
Already addressed the issue of the repeating lines. Merged the repeated lines.